### PR TITLE
fix race condition

### PR DESCRIPTION
**Pull request by ibreakthecloud**

### CODE REVIEW

File: pkg/scan/scanner.go [hunk 1]
```diff
@@ -86,11 +88,12 @@ func (s *Scanner) Scan(ctx *tasks.ScanContext, stype ScanType, namespace, id str
 	// results has to be 1 element max
 	// to avoid overwriting the buffer entries
 	results := make(chan []output.IOCFound)
-	defer close(results)
```
Comment (ibreakthecloud) [inline pkg/scan/scanner.go:89]: results will never close if function exits early :/

Comment (ibreakthecloud) [reply pkg/scan/scanner.go:89]: todo: fix this

**Comment by ibreakthecloud** [comment]
@noboruma PTAL

**Review by noboruma — APPROVED** [review]
No need for the wait group if we use a bool chan
Will address separately

**Pull request merged**